File: python-samples/log_line_substring_char_count_2.py
import sys

log_file_path = sys.argv[1]

char_count = 0

with open(log_file_path, 'r') as log_file:
    while line := log_file.readline():
        processed_line = line.rstrip('\n')
        request_on_index = processed_line.find("REQUEST on")
        if request_on_index != -1:
            substring = processed_line[request_on_index:].upper()
            for c in substring:
                if c == 'A' or c == 'E':
                    char_count += 1

print("char count:", char_count)

File: java-samples/src/main/java/posts/parthmistry/javasamples/LogLineSubstringCharCount2.java
package posts.parthmistry.javasamples;

import java.nio.file.Files;
import java.nio.file.Paths;

public class LogLineSubstringCharCount2 {

    public static void main(String[] args) throws Exception {
        var logFilePath = args[0];

        var charCount = 0L;

        try (var reader = Files.newBufferedReader(Paths.get(logFilePath))) {
            String line = null;
            while ((line = reader.readLine()) != null) {
                var requestOnIndex = line.indexOf("REQUEST on");
                if (requestOnIndex != -1) {
                    var substring = line.substring(requestOnIndex).toUpperCase();
                    for (int i = 0; i < substring.length(); i++) {
                        var c = substring.charAt(i);
                        if (c == 'A' || c == 'E') {
                            charCount += 1;
                        }
                    }
                }
            }
        }

        System.out.println("char count: " + charCount);
    }

}
